### PR TITLE
Crash in SubForm issue fixed

### PR DESCRIPTION
**Pull request by UmashriR**

### CODE REVIEW

File: ZCRMiOS/Classes/APIRequest.swift [hunk 1]
```diff
@@ -127,12 +138,22 @@ internal class APIRequest
...
+                return
+            }
             responseData = data
             urlResponse = response
             sema.signal()
         }).resume()
         sema.wait()
+        if error != nil
+        {
+            throw ZCRMSDKError.ConnectionError( "URLSession dataTask error : \( error!.description )" )
```
Comment (BoopathyPSiva) [inline ZCRMiOS/Classes/APIRequest.swift]: It should be ProcessingError

File: ZCRMiOS/Classes/ZCRMEventParticipant.swift [hunk 2]
```diff
@@ -49,7 +49,7 @@ public class ZCRMEventParticipant
     /// set the name of the ZCRMEventParticipant.
     ///
     /// - Parameter name: name of the ZCRMEventParticipant
-    internal func setName( name : String? )
+    internal func setName( name : String )
```
Comment (BoopathyPSiva) [inline ZCRMiOS/Classes/ZCRMEventParticipant.swift:52]: why did you change to non-optional?

**Review by vijayakrishnap — APPROVED** [review]
